broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java

```diff
@@ -1517,8 +1517,12 @@ private RemotingCommand getConsumeStats(ChannelHandlerContext ctx,
                     }
                 }
 
+                long pullOffset = this.brokerController.getConsumerOffsetManager().queryPullOffset(
+                    requestHeader.getConsumerGroup(), topic, i);
+
                 offsetWrapper.setBrokerOffset(brokerOffset);
                 offsetWrapper.setConsumerOffset(consumerOffset);
+                offsetWrapper.setPullOffset(pullOffset >= 0 ? pullOffset : 0);
```

Comment:
It needs to check if pullOffset is bigger than consumerOffset. Otherwise, inflight will be negative when the user resets offset.

Reply:
ok